### ISSUE TITLE
Drawer - Axe violation [Ensure controls are not nested as they are not announced by screen readers]

### ISSUE THREAD

**Issue by divyajindel** [Sage/carbon#4692]
### Current behaviour

Found an accessibility violation with Drawer --> with complex example
"Ensure controls are not nested as they are not announced by screen readers"

### Expected behaviour

there should be no Axe violation.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/dialog-full-screen--with-complex-example

### Carbon Version

latest

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by Parsium** [comment]
FE-4710
